### PR TITLE
enh(openid): log errors and improve logging message

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to log correctly External IDP Errors during authentication process through OpenID.
Previously the errors was logged only if a 200 was received (so indeed no errors...).
Also we now log the `error_description` and no more the `error` because `error_description` add way more context and precision than `error`

**Fixes** # MON-153171

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x
- [x] 24.10.x
- [x] master

<h2> How this pull request can be tested ? </h2>

Make an invalid OpenID Configuration (for example an invalid Client Secret).
Try to authenticate through OpenID.  Error logs should now appears in /var/log/centreon/centreon-web.log and /var/log/centreon/login.log

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 1]
```diff
@@ -999,10 +977,10 @@ private function logErrorForInvalidStatusCode(int $codeReceived, int $codeExpect
...
     private function logErrorInLoginLogFile(string $message, array $content): void
     {
-        if (array_key_exists('error', $content)) {
-            $this->centreonLog->insertLog(
-                CentreonUserLog::TYPE_LOGIN,
-                "[Openid] [Error] {$message}" . json_encode($content)
+        if (array_key_exists('error_description', $content)) {
+            CentreonLog::create()->error(
+                CentreonLog::TYPE_LOGIN,
+                "[Openid] {$message} " . $content['error_description']
```
Comment (Drago37) [inline centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:983]: erorr_description in the custom extra option

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 2]
```diff
@@ -1061,10 +1039,10 @@ private function logAuthenticationInfo(string $message, ?array $content = null):
...
-        $this->centreonLog->insertLog(
-            CentreonUserLog::TYPE_LOGIN,
+        CentreonLog::create()->error(
+            CentreonLog::TYPE_LOGIN,
             sprintf(
-                "[Openid] [Error] {$message}",
+                "[Openid] {$message}",
                 $exception::class,
                 $exception->getMessage()
             )
```
Comment (Drago37) [inline centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:1048]: you need to pass the exception on argument of the error method

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 3]
```diff
@@ -1045,9 +1023,9 @@ private function logAuthenticationDebug(string $message, array $content): void
      */
     private function logAuthenticationInfo(string $message, ?array $content = null): void
     {
-        $this->centreonLog->insertLog(
-            CentreonUserLog::TYPE_LOGIN,
-            "[openid] [INFO] {$message}" . ($content !== null ? ' : ' . json_encode($content) : '')
+        CentreonLog::create()->info(
+            CentreonLog::TYPE_LOGIN,
+            "[Openid] {$message}" . ($content !== null ? ' : ' . json_encode($content) : '')
```
Comment (Drago37) [inline centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:1028]: json_encode can be on error, you need to use the [JSON_THROW_ON_ERROR](https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error) flag to have an improved error handling ;)

Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:1028]: $content should be passed in argument of error() for the custom options

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 4]
```diff
@@ -838,9 +816,9 @@ private function getUsernameFromLoginClaim(): string
...
         }
         if (! array_key_exists($loginClaim, $this->userInformations)) {
-            $this->centreonLog->insertLog(
-                CentreonUserLog::TYPE_LOGIN,
-                '[Openid] [Error] Unable to get login from claim: ' . $loginClaim
+            CentreonLog::create()->error(
+                CentreonLog::TYPE_LOGIN,
+                '[Openid] Unable to get login from claim: ' . $loginClaim
             );
             $this->error('Login Claim not found', ['login_claim' => $loginClaim]);
```
Comment (Drago37) [inline centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:823]: two differents loggers, it's normal ?

Comment (jeremyjaouen) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:823]: Yes, there is the LoggerTrait that is been used to write in centreon-web.log and CentreonLog to write in login.log

Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:823]: ok thanks

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 5]
```diff
@@ -343,22 +335,19 @@ public function refreshToken(AuthenticationTokens $authenticationTokens): Authen
 
         // Get the status code and throw an Exception if not a 200
         $statusCode = $response->getStatusCode();
+        $content = json_decode($response->getContent(false), true) ?: [];
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: You can use named parameters to not use the optionnal parameters before the flag ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 6]
```diff
@@ -495,22 +484,19 @@ private function sendRequestForConnectionTokenOrFail(string $authorizationCode,
...
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: You can use named parameters to not use the optionnal parameters before the flag ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 7]
```diff
@@ -630,22 +616,18 @@ private function sendRequestForIntrospectionEndpoint(): array
         }
 
         $statusCode = $response->getStatusCode();
+        $content = json_decode($response->getContent(false), true) ?: [];
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: You can use named parameters to not use the optionnal parameters before the flag ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 8]
```diff
@@ -703,22 +685,18 @@ private function sendRequestForUserInformationEndpoint(): array
...
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: You can use named parameters to not use the optionnal parameters before the flag ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 9]
```diff
@@ -1030,9 +1008,9 @@ private function logAuthenticationDebug(string $message, array $content): void
         if (isset($content['provider_token'])) {
             $content['provider_token'] = mb_substr($content['provider_token'], -10);
         }
-        $this->centreonLog->insertLog(
-            CentreonUserLog::TYPE_LOGIN,
-            "[Openid] [Debug] {$message} " . json_encode($content)
+        CentreonLog::create()->debug(
+            CentreonLog::TYPE_LOGIN,
+            "[Openid] {$message} " . json_encode($content)
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: ok

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 1 above)
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php:983]: I am not agree with this practice but I resolve.

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 2 above)
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: You can pass the exception one time using the last argument. You don't need to pass the exception in the extra ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 3 above)
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: ok

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: Seen together, I resolve ! The next time, you can use the named parameters ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: Seen together, I resolve ! The next time, you can use the named parameters ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: Seen together, I resolve ! The next time, you can use the named parameters ;)

File: centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php (hunk 2 above)
Comment (Drago37) [reply centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: Seen together, I resolve ! The next time, you can pass the exception in the extra options ;)

**Review by Drago37 — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**